renderer/brave_content_renderer_client.cc

```diff
@@ -63,6 +65,27 @@
 #endif
 
 namespace {
+
+void EnsureBraveInternalWorldInitialized() {
+  static bool initialized = false;
+  if (initialized) {
+    return;
+  }
+
+  initialized = true;
+
+  // Set an empty CSP so that the main world's CSP is not used in the isolated
+  // world.
+  constexpr char kContentSecurityPolicy[] = "";
+
+  blink::WebIsolatedWorldInfo info;
+  info.security_origin =
+      blink::WebSecurityOrigin::Create(GURL("chrome://brave_internal"));
```

Comment:
why do we have to set a security origin?

Reply:
actually i see in the docs that this is needed if the CSP is set, but it's not set in Brave Translate which also sets a CSP? https://github.com/brave/brave-core/blob/ea08eba2f949b38917742c32707ebf30aa859395/chromium_src/components/translate/content/renderer/isolated_world_util.cc#L17

Reply:
It's a method to adjust the upstream policy. It reuses the security origin is set here:
https://source.chromium.org/chromium/chromium/src/+/main:components/translate/content/renderer/isolated_world_util.cc;l=34-35

Reply:
This seems like a good change, but maybe it should be a separate PR now since we're not doing anything on the renderer side anymore in this PR?

Reply:
I believe the isolated world is automatically created the first time you attempt to execute JS. We have lots of other JS execution in isolated worlds outside of cosmetic filtering. Are you sure it doesn't work without this?

Reply:
For the twitter script, we're using the same technique as cosmetic filters (inserting a script tag in order to execute a small bit of JS in the main world), which requires the CSP customization.

Reply:
@diracdeltas I suppose that we could introduce a new world ID and unconditionally initialize it (i.e. initialize it for all renderers) similar to what we're doing here with `ISOLATED_WORLD_ID_BRAVE_INTERNAL`.

The current state, though, is that everything using `ISOLATED_WORLD_ID_BRAVE_INTERNAL` is using the custom CSP (as long as the cosmetic filters feature is not disabled, that is). Seems a bit out of scope to rework isolated worlds here.

I could also just back out this change and let the Twitter detection fail if cosmetic filters is disabled. Would that be preferrable?

Reply:
> I could also just back out this change and let the Twitter detection fail if cosmetic filters is disabled. Would that be preferable?

yeah i think we should do that for this PR and have a separate issue/PR for moving cosmetic filters to its own world and twitter detection to its own world

Reply:
I've removed the changes on the renderer side.